overlord/snapstate/snapstate.go

```diff
@@ -2000,6 +2003,36 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 		}
 	}
 
+	// if specific gating snap was given, drop other affected snaps unless
+	// they are affected by same updates as forGatingSnap.
+	if forGatingSnap != "" {
+		if _, ok := affectedSnaps[forGatingSnap]; !ok {
+			// this can happen if after refreshing candidates the snap is no
+			// longer affected by new candidates.
+			return nil, nil, nil
+		}
+		// filter out unrelated affectedSnaps
+		for gatingSnap, affectedInfo := range affectedSnaps {
+			if gatingSnap == forGatingSnap {
+				continue
+			}
+			var found bool
+			for affectingSnap := range affectedInfo.AffectingSnaps {
+				// does this affectingSnap also affect forGatingSnap?
+				// if so, we will keep it around, meaning we will run its
```

Comment:
iiuc what we keep around is the full gatinSnap (that's what the delete below is about), so this comment is a bit confusing and hard to parse.  The condition is really that we keep gating/affectedSnap around if the intersection between its affecting snaps and the one for forGatingSnap is not empty, that means we should call its hook because we can't update the affecting snaps if they affect something else. Which I think is what the code here does correctly, but the comment doesn't explain quite clearly.

Reply:
Yes, your understanding is correct. I've reworded this comment, hope it makes more sense. Thanks